src/js/pages/services/DeploymentsTab.js

```diff
@@ -1,73 +1,158 @@
+import classNames from 'classnames';
+import {Link} from 'react-router';
 import moment from 'moment';
+import mixin from 'reactjs-mixin';
+/* eslint-disable no-unused-vars */
 import React from 'react';
+/* eslint-enable no-unused-vars */
+import {StoreMixin} from 'mesosphere-shared-reactjs';
 import {Table} from 'reactjs-components';
 
 import AlertPanel from '../../components/AlertPanel';
-import EventTypes from '../../constants/EventTypes';
-import MarathonStore from '../../stores/MarathonStore';
+import DCOSStore from '../../stores/DCOSStore';
 import ResourceTableUtil from '../../utils/ResourceTableUtil';
-import StringUtil from '../../utils/StringUtil'
+import ServicesBreadcrumb from '../../components/ServicesBreadcrumb';
+import StringUtil from '../../utils/StringUtil';
 
 const columnClassNameGetter = ResourceTableUtil.getClassName;
 const columnHeading = ResourceTableUtil.renderHeading({
-  id: 'DEPLOYMENT',
+  id: 'AFFECTED SERVICES',
   startTime: 'STARTED',
+  location: 'LOCATION',
   status: 'STATUS'
 });
-const columns = [
-  {
-    className: columnClassNameGetter,
-    heading: columnHeading,
-    prop: 'id',
-    render: function (prop, deployment) {
-      return deployment.getId();
-    }
-  },
-  {
-    className: columnClassNameGetter,
-    heading: columnHeading,
-    prop: 'startTime',
-    render: function (prop, deployment) {
-      return moment(deployment.getStartTime()).fromNow();
-    }
-  },
-  {
-    className: columnClassNameGetter,
-    heading: columnHeading,
-    prop: 'status',
-    render: function (prop, deployment) {
-      return `Step ${deployment.getCurrentStep()} of ${deployment.getTotalSteps()}`;
-    }
-  }
+const METHODS_TO_BIND = [
+  'renderAffectedServices',
+  'renderAffectedServicesList',
+  'renderLocation',
+  'renderStartTime',
+  'renderStatus'
 ];
 
-class DeploymentsTab extends React.Component {
+class DeploymentsTab extends mixin(StoreMixin) {
 
   constructor() {
     super(...arguments);
-    this.onDeploymentsChange = this.onDeploymentsChange.bind(this);
 
-    this.state = {
-      deployments: MarathonStore.get('deployments')
-    };
+    this.state = {deployments: DCOSStore.deploymentsList};
+    this.store_listeners = [
+      {name: 'dcos', events: ['change'], suppressUpdate: true}
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    }, this);
+  }
+
+  onDcosStoreChange() {
+    this.setState({deployments: DCOSStore.deploymentsList});
```

Comment:
Happy to respect the convention if it's genuinely preferred, but I'd like to understand what the benefit is of breaking with React idiom. Forcing updates via the store mixin makes it hard for me to understand the update/render cycle and circumvents `shouldComponentUpdate`.